frontend/src/notifications.tsx

```diff
@@ -0,0 +1,56 @@
+import * as React from "react";
+import { Button } from "@chakra-ui/react";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faBellSlash } from "@fortawesome/free-solid-svg-icons";
+import { isEqual } from "lodash-es";
+
+type KeyType = number | string;
+
+type NotificationOptions<T> = {
+  filter: (x: T) => boolean;
+  message: (xs: T[]) => string;
+  key: (x: T) => KeyType;
+};
+
+export function useNotification<T>(
+  xs: T[],
+  { filter, message, key }: NotificationOptions<T>
+) {
+  const [seen, setSeen] = React.useState<KeyType[]>([]);
+  const filtered = xs.filter(filter);
+  const unseen = filtered.filter((x) => !seen.includes(key(x)));
+  React.useEffect(() => {
+    if (unseen.length > 0) {
+      const msg = message(unseen);
+      if (Notification.permission === "granted") {
+        new Notification(msg);
+        new Audio("/sounds/bell.mp3").play();
```

Comment:
Woah, notifications don't have their own system-configured sounds?

Reply:
I don't really know. I didn't notice a sound when it popped up, though.